WordPress/gutenberg
# Show Link UI in Nav Offcanvas for submenus inserted via block options ("3 dots") menu

## What?


Displays the Link UI for submenus inserted via the block options ("3 dots") menu in the Nav Offcanvas experiment.

⚠️ Note: this does _not_ currently work for the case where the given block is _not_ currently a submenu (i.e. it's a "Custom Link"). Why? Because the relevant code calls `replaceBlock` which (unlike `insertBlock`) does not currently have an argument to disable the select of the "replaced" block in the canvas 🤦 

https://github.com/WordPress/gutenberg/blob/aa9b1d665a10f32182797e0d7e7a559362d1ec1c/packages/block-editor/src/components/off-canvas-editor/block.js#L384-L397

## Why?


Because when a new link is inserted the Link UI should be displayed to afford the ability to configure the link in the same way in which the UI is displayed when adding a link via the block appender `+`.

## How?


This PR:

- lifts state up to the offcanvas root.
- passes `onInsertBlock` callback down to appender and individual block components
- components call `onInsertBlock` anytime a block is inserted to trigger the Link UI
- handling of information about the "inserted block" is now extracted to a hook for utility

## Testing Instructions





- Test everything behaves as per `trunk` for normal link insertion via appender
- Now with a Custom Link click on the options menu ("3 dots menu") and click `Add submenu item`.
- See that the item is inserted by the Link UI does _not_ show (this is due to the bug described under `What` above which we will address in a follow up to target the missing APIs on `replaceBlock`.
- Now you have a Submenu item click on the options menu ("3 dots menu") and click `Add submenu item`.
- This time you should see the Link UI appear.

Also run the unit tests for the custom hook:

```
npm run test:unit packages/block-editor/src/components/off-canvas-editor/test/use-inserted-block.js
```

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://user-images.githubusercontent.com/444434/207874046-30efed05-0fef-42a8-87cc-0099df2c161c.mp4
